sway/input/keyboard.c

```diff
@@ -468,6 +469,7 @@ static void handle_key_event(struct sway_keyboard *keyboard,
 
 	if (binding) {
 		seat_execute_command(seat, binding);
+		transaction_commit_dirty();
```

Comment:
What was the motivation behind the changes in `keyboard.c`? Does `keyboard_execute_compositor_binding` already cause commits internally? (If so, perhaps making `seat_execute_command` or one of its callees fire the commits makes things more symmetrical?)

In general it'd be nice to push the transaction committing as low as possible, to reduce the probability of messing up in new code. If all `seat_execute_command`s must be followed by a `transaction_commit_dirty`, I think the commands should do that themselves.

Reply:
The reason to do it higher up is to combine multiple operations into a single transaction. Moving it down is a more significant change in behavior.

I don't personally have any attachment to that behavior, and would indeed find it less error prone to move it down, but whether we want that change should be considered.

(Even with the amend logic, the first transaction commits immediately.)

Reply:
I moved the calls from keyboard, but the seatop ones are trickier. They call `arrange_*` and container updates arbitrarily, so we can't easily move the commit down there without causing transactions to be broken up as mentioned above.

Reply:
> Compositor bindings are just the hardcoded VT change keybind.

Ack.

This sounds like a good middle ground, thanks.